components/p3a/brave_p3a_star_randomness_meta.cc

```diff
@@ -0,0 +1,278 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/p3a/brave_p3a_star_randomness_meta.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/base64.h"
+#include "base/bind.h"
+#include "base/json/json_reader.h"
+#include "base/logging.h"
+#include "brave/components/nitro_utils/attestation.h"
+#include "brave/components/p3a/brave_p3a_config.h"
+#include "brave/components/p3a/network_annotations.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "services/network/public/mojom/url_response_head.mojom.h"
+
+namespace brave {
+
+namespace {
+
+constexpr char kCurrentPKPrefName[] = "brave.p3a.current_pk";
+constexpr char kCurrentEpochPrefName[] = "brave.p3a.current_epoch";
+constexpr char kNextEpochTimePrefName[] = "brave.p3a.next_epoch_time";
+constexpr std::size_t kMaxInfoResponseSize = 131072;
```

Comment:
This could be more clearly written `128 * 1024 * 1024`. Also please comment that you chose the value as a generous arbitrary limit so future readers know where the number came from.